crates/sui-core/src/authority_active/execution_driver/mod.rs

```diff
@@ -63,6 +63,11 @@ where
 
         debug!("Pending certificate execution activated.");
 
+        // Process any tx that failed to commit.
+        if let Err(err) = active_authority.state.process_tx_recovery_log(None).await {
```

Comment:
Are we sure this is thread / multi task safe? Because in the past this was only intented for use at start up. Should we simply not modify the `acquire_tx_guard` to also return a valid handle if a past instance was abandonned?

Reply:
`recoverable_txes` is behind a mutex, so it should be thread safe.
We could modify `acquire_tx_guard` to just return a valid handle, but how is that different from this? We would just be moving the call to into `acquire_tx_guard`, which would run in parallel with other tasks anyway